jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java

```diff
@@ -149,8 +108,43 @@ static Arch detectArch() {
     }
   }
 
+  /**
+   * Utility class to determine the path of a native library.
+   */
+  static class LibraryPathProvider {
```

Comment:
Is there a reason why this is a separate class rather than adding these fields & methods to `NativeLibLoader`?

Reply:
I think it has a clear purpose, reduces the scope for `nativeResourcePath` and `customNativeLibraryPathPropertyPrefix` and aligns nicely with the corresponding test.

Reply:
@Craigacp Is this OK with you?

Reply:
I think spinning a class file for two fields and two methods isn't worth the effort, but it also doesn't really matter either way. The new test file needs a license header, and aside from that everything looks ok.

Reply:
Good point! I've added the license info.

(b4954f81)